### PR TITLE
Add documentation for building and running Charm programs with UCX backend

### PR DESCRIPTION
**Pull request by nitbhat**

### CODE REVIEW

File: doc/charm++/manual.rst [hunk 1]
```diff
@@ -10880,6 +10880,83 @@ The charm directory contains a collection of example-programs and
...
+with the UCX backend using different process management interfaces. In all these cases,
+in order for Charm++ to find UCX path, it is required to pass the UCX build directory
+as ``--basedir``.
+
+To build the Charm++ target with Simple PMI, do not specify any additional option as shown
+in the build command.
+
+.. code-block:: bash
+
+   $ ./build Charm++ ucx-linux-x86_64 --with-production --enable-error-checking --basedir=$HOME/ucx/build -j16
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: I think we usually keep charm++ in lower case in the build line.

File: doc/charm++/manual.rst [hunk 2]
```diff
@@ -10880,6 +10880,83 @@ The charm directory contains a collection of example-programs and
...
+
+   $ ./build Charm++ ucx-linux-x86_64 slurmpmi2 --with-production --enable-error-checking --basedir=$HOME/ucx/build -j16
+
+To build the Charm++ target with PMIx, you would require an OpenMPI implementation with PMIx
+enabled to be installed on your system. In case OpenMPI is not available in your environment,
+you can build OpenMPI from scratch using the following steps:
+
+.. code-block:: bash
+
+  wget https://download.open-mpi.org/release/open-mpi/v4.0/openmpi-4.0.1.tar.gz
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: could add '$' here.

File: doc/charm++/manual.rst [hunk 3]
```diff
@@ -10880,6 +10880,83 @@ The charm directory contains a collection of example-programs and
...
+   $ make -j8
+   $ make install
+
+After installing UCX, there are several supported process management interfaces (PMI)
+that can be specified as options in order to build Charm++ with UCX. These include
+Simple PMI, Slurm PMI, Slurm PMI 2 and PMIx (using OpenMPI). Currently, in order to
+use PMIx for process management, it is required to have OpenMPI installed on the system.
+Additionally, in order to use the other supported process management interfaces, it is
+required to have a non-OpenMPI based MPI implementation installed on the system (e.g.
+Intel MPI, Mvapich, MPICH, etc.).
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: could capitalize Mvapich

File: doc/charm++/manual.rst [hunk 4]
```diff
@@ -10880,6 +10880,83 @@ The charm directory contains a collection of example-programs and
...
+
+The following section shows examples of build commands that can be used to build targets
+with the UCX backend using different process management interfaces. In all these cases,
+in order for Charm++ to find UCX path, it is required to pass the UCX build directory
+as ``--basedir``.
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: maybe add here something like 'unless UCX is installed in a system directory'?

File: doc/charm++/manual.rst [hunk 5]
```diff
@@ -11646,6 +11723,20 @@ directory. Pathname resolution is performed as follows:
...
+
+Instructions to run Charm++ programs for Specific Builds
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+UCX
+^^^
+When Charm++ is built with UCX using Simple PMI, Slurm PMI or Slurm PMI2, ensure that
+an MPI implementation which is not OpenMPI is loaded in the environment. Use the mpiexec/mpirun
+launcher provided by the non-OpenMPI implementation to launch programs. Alternatively, you
+can also use any system provided launchers.
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: maybe add here:  'Note that due to bug #2477, binaries built with UCX cannot be run without mpiexec/mpirun.'

File: doc/charm++/manual.rst [hunk 6]
```diff
@@ -11646,6 +11723,20 @@ directory. Pathname resolution is performed as follows:
...
    appended extension if specified) on all nodes.
 
+
+Instructions to run Charm++ programs for Specific Builds
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+UCX
+^^^
+When Charm++ is built with UCX using Simple PMI, Slurm PMI or Slurm PMI2, ensure that
+an MPI implementation which is not OpenMPI is loaded in the environment. Use the mpiexec/mpirun
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: might be sufficient to only mention mpirun instead of mpiexec/mpirun.

File: doc/charm++/manual.rst (hunk 3 above)
Comment (nitbhat) [reply doc/charm++/manual.rst]: Done.

File: doc/charm++/manual.rst (hunk 4 above)
Comment (nitbhat) [reply doc/charm++/manual.rst]: Done.

File: doc/charm++/manual.rst (hunk 5 above)
Comment (nitbhat) [reply doc/charm++/manual.rst]: Done.

File: doc/charm++/manual.rst (hunk 6 above)
Comment (nitbhat) [reply doc/charm++/manual.rst]: Done.

File: doc/charm++/manual.rst (hunk 2 above)
Comment (nitbhat) [reply doc/charm++/manual.rst]: Done.

File: doc/charm++/manual.rst (hunk 1 above)
Comment (nitbhat) [reply doc/charm++/manual.rst]: Good point, done.

**Comment by nitbhat** [comment]
Thanks Matt, for highlighting some really good points in the review.

**Pull request merged**